editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx

```diff
@@ -84,44 +84,194 @@ describe("Basic UI", () => {
+    await user.click(continueButton);
+
+    const errorMessages = getAllByTestId(/error-message-input/);
+
+    // 5 error message inputs + 3 for date inputs (one per input and one wrapper)
+    expect(errorMessages).toHaveLength(4);
+
+    errorMessages.forEach((message) => {
+      expect(message).not.toBeEmptyDOMElement();
+    });
+  });
+
+  // it shows all fields are required in a tab
+  it("should show all fields are required, for all feature tabs", async () => {
+    const { getByTestId, getByRole, user, debug } = setup(
+      <MapAndLabel {...props} />,
+    );
+    const map = getByTestId("map-and-label-map");
+    expect(map).toBeInTheDocument();
+    debug();
+
+    addMultipleFeatures([point1, point2]);
+
+    // vertical side tab query
+    const firstTab = getByRole("tab", { name: /Tree 1/ });
+    const secondTab = getByRole("tab", { name: /Tree 2/ });
+
+    // side tab validation
+    expect(firstTab).toBeInTheDocument();
+    expect(secondTab).toBeInTheDocument();
+
+    // form for each tab
+    const firstTabPanel = getByTestId("vertical-tabpanel-0");
+    const secondTabPanel = getByTestId("vertical-tabpanel-1");
+
+    // default is to start on seond tab panel, second is hidden
+    expect(firstTabPanel).not.toBeVisible();
+    expect(secondTabPanel).toBeVisible();
+
+    // Form is generate for secondTabPanel but not the first
+    expect(secondTabPanel.childElementCount).toBeGreaterThan(0);
+    expect(firstTabPanel.childElementCount).toBe(0);
+
+    const continueButton = getByRole("button", { name: /Continue/ });
+    await user.click(continueButton);
+
+    // error messages appear
+    const errorMessagesTabTwo =
+      within(secondTabPanel).getAllByTestId(/error-message-input/);
+    expect(errorMessagesTabTwo).toHaveLength(4);
+
+    // error messages are empty but visible before error state induced
+    // this ensures they contain the error message text
+    errorMessagesTabTwo.forEach((input) => {
+      expect(input).not.toBeEmptyDOMElement();
+    });
+
+    // click to go to the second tab
+    await user.click(firstTab);
+
+    expect(firstTabPanel).toBeVisible();
+
+    // error messages persist
+    const errorMessagesTabOne =
+      within(firstTabPanel).getAllByTestId(/error-message-input/);
+    expect(errorMessagesTabOne).toHaveLength(4);
+  });
+
+  // it shows all fields are required across different tabs
+  it("should show an error if the minimum number of items is not met", async () => {
+    const { getByTestId, getByRole, user } = setup(<MapAndLabel {...props} />);
+    const map = getByTestId("map-and-label-map");
+    expect(map).toBeInTheDocument();
+
+    const continueButton = getByRole("button", { name: /Continue/ });
+
+    await user.click(continueButton);
+
+    const errorWrapper = getByTestId(/error-wrapper/);
+
+    const errorMessage = within(errorWrapper).getByText(/You must plot /);
+
+    expect(errorMessage).not.toBeEmptyDOMElement();
```

Comment:
nit: I think `toBeInTheDocument()` would be a stronger check _for visibility_ here? Per a11y requirements, we have many error wrappers that are _always_ in the DOM, and their _visibility_ is toggled depending on component state.

Reply:
If this suggested change worked as expected here - can you also please update all other similar instances of `DOMElement` checks as well - otherwise I think these are quite flakey tests _not_ actually testing _visibility_ of error messages like we intend !!

Reply:
@jessicamcinchak for this, would it be the case that for the Input fields in the `<TabPanels>` that the errors are always in the document but just don't have any text in them so `toBeInTheDocument()` would come back yes even if there's no error messages displaying? 

Would something like this be more specific?

```ts
    errorMessages.forEach((message) => {
      expect(message).toHaveTextContent(/(Enter|Select)/);
    });
```

Reply:
Example here of error element before "Continue" is clicked:

```html

<p class="..." id="error-message-input-text-species" data-testid="error-message-input-text-species"></p>
```

Reply:
Looking through other error handling tests in other components now and realising checking DOM element is used elsewhere too, let's leave this as-is right now. Sorry for confusion. 

There's still many outstanding to-do's here that are more valuable to move onto than get hung up on this.